tests/commands/create_project.py

```diff
@@ -0,0 +1,269 @@
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import pytest
+
+from django.core.exceptions import ValidationError
+from django.core.management import call_command, CommandError
+
+from pootle.core.plugin import provider
+from pootle_fs.delegate import fs_plugins
+from pootle_fs.plugin import Plugin
+from pootle_project.models import Project
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_defaults(settings):
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.fullname == "Foo0"
+    assert foo0.checkstyle == "standard"
+    assert foo0.filetypes.first().name == "po"
+    assert foo0.source_language.code == "en"
+    assert foo0.disabled is False
+    assert foo0.report_email == ""
+
+    assert (
+        foo0.config["pootle_fs.translation_mapping"]
+        == dict(default=u'/po/<language_code>.<ext>'))
+    assert foo0.config["pootle_fs.fs_type"] == "localfs"
+    assert (
+        foo0.config["pootle_fs.fs_url"]
+        == ("{POOTLE_TRANSLATION_DIRECTORY}%s"
+            % foo0.code))
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_existing_project(settings):
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=gnu")
+
+    with pytest.raises(ValidationError):
+        call_command(
+            "create_project",
+            "foo0",
+            "--preset-mapping=gnu")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_title(capfd):
+    call_command(
+        "create_project",
+        "foo1",
+        "--name=Bar1",
+        "--preset-mapping=gnu")
+    foo1 = Project.objects.get(code="foo1")
+    assert foo1.fullname == "Bar1"
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_checkstyle(capfd):
+    call_command(
+        "create_project",
+        "foo0",
+        "--checkstyle=minimal",
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.checkstyle == "minimal"
+
+    with pytest.raises(CommandError):
+        call_command(
+            "create_project",
+            "foo1",
+            "--preset-mapping=gnu",
+            "--checkstyle=DOESNOTEXIST")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_filetypes(capfd):
+    call_command(
+        "create_project",
+        "foo0",
+        "--filetype=ts",
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.filetypes.first().name == "ts"
+
+    call_command(
+        "create_project",
+        "foo1",
+        "--filetype=ts",
+        "--preset-mapping=gnu")
+    foo1 = Project.objects.get(code="foo1")
+    assert foo1.filetypes.first().name == "ts"
```

Comment:
Why two valid calls that are essentially the same?

Reply:
ill remove one of these